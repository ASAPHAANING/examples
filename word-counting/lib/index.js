const fs = require('fs');
const path = require('path');
const bench = require('./bench');
const js = require('./search');
const neon = require('../native/index.node');

const wc = {
  js,
  neon
};

const ROOT = path.resolve(__dirname, '..');
const DATA = path.resolve(ROOT, 'data');

const string = fs.readFileSync(
  path.resolve(DATA, 'shakespeare-plays.csv'),
  'utf8'
);
const buffer = fs.readFileSync(path.resolve(DATA, 'shakespeare-plays.csv'));

console.log(
  'Node:    ',
  bench(() => wc.js.search(string, 'thee'))
);
console.log(
  'Neon:          ',
  bench(() => wc.neon.search(buffer, 'thee'))
);
console.log(
  'Neon(parallel):',
  bench(() => wc.neon.search(buffer, 'thee', true))
);

module.exports = neon;
