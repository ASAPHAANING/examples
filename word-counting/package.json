{
  "name": "word-counting-demo",
  "description": "A word count demo to showcase Neon.",
  "main": "lib/index.js",
  "dependencies": {
    "neon-cli": "^0.2.0"
  },
  "scripts": {
    "install": "neon build --release"
  }
}
