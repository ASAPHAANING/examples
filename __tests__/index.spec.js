const helloWorld = require('../hello-world');
const threadCount = require('../thread-count');
const primitives = require('../primitives');
const arrays = require('../arrays');
const objects = require('../objects');
const args = require('../arguments');
const functions = require('../functions');
const fibonacciTask = require('../fibonacci-async-task');
const sharingBinaryData = require('../sharing-binary-data');
const json = require('../json');
const classes = require('../classes');
const modules = require('../modules');
const errors = require('../errors');
const async = require('../async');
const Emitter = require('../event-emitter/lib/emitter');
require('../word-counting');

describe('tests', () => {
  it('should run hello world', () => {
    expect(helloWorld.helloWorld()).toEqual('hello world!');
  });

  it('should run thread count', () => {
    expect(Number.isInteger(threadCount.threadCount())).toEqual(true);
  });

  it('should run primitives', () => {
    primitives.primitives();
  });

  it('should run arrays', () => {
    expect(arrays.convertVecToArray()).toEqual([]);
    expect(arrays.returnJsArray()).toEqual([]);
    expect(arrays.returnJsArrayWithNumber()).toEqual([9000]);
    expect(arrays.returnJsArrayWithString()).toEqual(['hello node']);
  });

  it('should run objects', () => {
    expect(objects.convertStructToJsObject()).toEqual({
      myNumberProperty: 1234,
      myStringProperty: 'baz'
    });
  });

  it('should run arguments', () => {
    args.printFunction(() => {});
    expect(args.add1ToArgument(1)).toEqual(2);
    expect(args.getArgsLen(1, 2, 3)).toEqual(3);
    args.argsOpt(1);
    args.defaultArgs();
  });

  it('should run functions', () => {
    expect(functions.requireObjectThis()).toEqual(undefined);
    expect(functions.callJsFunction(() => 12)).toEqual(12);
    expect(functions.constructJsFunction(Date)).toEqual(1970);
    expect(functions.returnJsFunction()(1)).toEqual(2);
  });

  it('should run json', () => {
    expect(json.sayHello('should run john')).toEqual('Hello, should run john!');
  });

  it('should run share binary data', () => {
    expect(new Float32Array(sharingBinaryData.hello())).toEqual(
      new Float32Array([0, 0])
    );
  });

  it('should run classes', () => {
    const { User } = classes;
    new User(0, 'John', 'Doe', 'johndoe@gmail.com');
  });

  it('should run fibonacci-task', async () => {
    const fib10 = await new Promise((resolve, reject) => {
      fibonacciTask.fibonacci(10, (err, result) => {
        if (err) reject(err);
        resolve(result);
      });
    });
    expect(fib10).toEqual('55');
    expect(fibonacciTask.fibonacciSync(10)).toEqual('55');
  });

  it('should run modules', () => {
    expect(modules.hello()).toEqual('hello node');
    new modules.User(0, 'Jane', 'Doe', 'janedoe@gmail.com');
    expect(modules.baz).toEqual('baz');
  });

  it('should run errors', () => {
    expect(() => errors.throwError()).toThrow('');
    expect(() => errors.throwCustomError('type_error')).toThrow(
      'throwing a TypeError'
    );
  });

  it('should run async', async () => {
    const result = await new Promise(resolve => {
      async.performAsyncTask((err, value) => resolve(value));
    });
    expect(result).toEqual(17);
  });

  it('should run event emitter', () => {
    const e = new Emitter();
    const spy = jest.fn();
    e.on('tick', spy);
    expect.assertions(2);
    setTimeout(() => {
      e.shutdown();
    }, 1250);
    return new Promise(resolve => {
      setTimeout(() => {
        expect(spy).toHaveBeenCalledTimes(2);
        expect(spy).toHaveBeenLastCalledWith({ count: 1 });
        resolve();
      }, 1750);
    });
  });
});
