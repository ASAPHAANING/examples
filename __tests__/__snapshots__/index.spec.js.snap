// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests should run arguments 1`] = `[Function]`;

exports[`tests should run arguments 2`] = `2`;

exports[`tests should run arguments 3`] = `3`;

exports[`tests should run arguments 4`] = `undefined`;

exports[`tests should run arguments 5`] = `undefined`;

exports[`tests should run arrays 1`] = `Array []`;

exports[`tests should run arrays 2`] = `Array []`;

exports[`tests should run arrays 3`] = `
Array [
  9000,
]
`;

exports[`tests should run arrays 4`] = `
Array [
  "hello node",
]
`;

exports[`tests should run async 1`] = `55`;

exports[`tests should run classes 1`] = `User {}`;

exports[`tests should run errors 1`] = `[Function]`;

exports[`tests should run errors 2`] = `[Function]`;

exports[`tests should run event emitter 1`] = `
Object {
  "a": 1,
  "b": Object {
    "c": Array [
      2,
      3,
      Object {
        "d": "4",
      },
    ],
  },
}
`;

exports[`tests should run event emitter 2`] = `
Object {
  "a": 1,
  "b": Object {
    "c": Array [
      2,
      3,
      Object {
        "d": "4",
      },
    ],
  },
}
`;

exports[`tests should run event emitter 3`] = `
Object {
  "a": 1,
  "b": Object {
    "c": Array [
      2,
      3,
      Object {
        "d": "4",
      },
    ],
  },
}
`;

exports[`tests should run fibonacci-task 1`] = `"55"`;

exports[`tests should run fibonacci-task 2`] = `"55"`;

exports[`tests should run functions 1`] = `undefined`;

exports[`tests should run functions 2`] = `12`;

exports[`tests should run functions 3`] = `1970`;

exports[`tests should run functions 4`] = `[Function]`;

exports[`tests should run hello world 1`] = `"hello world!"`;

exports[`tests should run json 1`] = `"Hello, should run john!"`;

exports[`tests should run modules 1`] = `[Function]`;

exports[`tests should run modules 2`] = `[Function]`;

exports[`tests should run modules 3`] = `"baz"`;

exports[`tests should run objects 1`] = `
Object {
  "myNumberProperty": 1234,
  "myStringProperty": "baz",
}
`;

exports[`tests should run primitives 1`] = `undefined`;

exports[`tests should run share binary data 1`] = `
Float32Array [
  0,
  0,
]
`;
